src/gtc/dace/nodes.py

```diff
@@ -40,32 +40,58 @@ def _set_expansion_order(
     node._expansion_specification = res
 
 
+class PickledProperty:
+    def to_json(self, obj):
+        protocol = pickle.DEFAULT_PROTOCOL
+        pbytes = pickle.dumps(obj, protocol=protocol)
+        jsonobj = dict(pickle=base64.b64encode(pbytes).decode("utf-8"))
+        return jsonobj
+
+    @classmethod
+    def from_json(cls, d, sdfg=None):
+        b64string = d["pickle"]
+        byte_repr = base64.b64decode(b64string)
+        return pickle.loads(byte_repr)
```

Comment:
Not something that directly affects this PR, but just a thought: is pickling the right approach for serializing SDFGs? AFAIK It's not safe between versions, and is a direct memory representation. Perhaps something like JSON directly is better?

Reply:
Yes, this is a workaround for a limitation similar to https://github.com/GridTools/gt4py/issues/352 (AFAIK still not possible) even if the particular example in the issue may be fixed.